### PR TITLE
community-ci: declare "early CI"

### PR DESCRIPTION
**Pull request by alafanechere**
## What
We want to run some CI checks on community PRs automatically.
For safety reasons these checks would be the one which:
* Do not require secrets
* Do not require connector code execution

This PR makes the static checks (a.k.a qa checks) and version increment checks run automatically on community PRs.

## How
* Declare an "early CI" job in the community ci workflow
* Run `airbyte-ci connectors test --only-step=qa_checks --only-step=version_inc_check`
* Modify airbyte-ci to customize the name of the global status check that are emitted by a test command

### CODE REVIEW

**Comment by alafanechere** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37690?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#37690** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37690?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#37683** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37683?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/community_ci.yml [hunk 1]
```diff
@@ -42,12 +42,13 @@ jobs:
...
       # This will sync the .github folder of the main repo with the fork
       # This allows us to use up to date actions from the main repo
-      - name: Pull .github folder from main repository
+      - name: Pull .github folder and internal packages from main repository
         id: pull_github_folder
         run: |
           git remote add main https://github.com/airbytehq/airbyte.git
           git fetch main ${MAIN_BRANCH_NAME}
           git checkout main/${MAIN_BRANCH_NAME} -- .github
+          git checkout main/${MAIN_BRANCH_NAME} -- airbyte-ci
```
Comment (natikgadzhi) [inline .github/workflows/community_ci.yml:51]: This will make it impossible to test any changes to airbyte-ci itself, and I don't think this is necessary.

**Review by natikgadzhi — APPROVED** [review]
I love that we're improving, and I love pre-ci idea. Approving to move forward. 

That said, two pieces of feedback I have for you are: 
- Please do not discard all PRs to CDK and to airbyte-ci. I don't think you have to do this. Perhaps we can only do this for pre-ci, but keep airbyte-ci directory from fork for full CI, so we get security, but keep ability to test airbyte-ci changes on demand with approval.
- Ditto formatters, while separate PR, formatters should also run without priviledges and automatically without human approval I think.

**Comment by alafanechere** [comment]
> Please do not discard all PRs to CDK and to airbyte-ci. I don't think you have to do this. Perhaps we can only do this for pre-ci, but keep airbyte-ci directory from fork for full CI, so we get security, but keep ability to test airbyte-ci changes on demand with approval

I'm thinking of creating a different workflow for CDK and airbyte-ci contributions.
My plan is to restructure our workflow to reduce the number of workflow files to simplify maintainance and shared logic.

I'd ideally like to have:
* A connector specific workflow which would run all required checks for connectors PRs, both on forks and not on forks
* An internal packages specific workflow

**Pull request merged**